R/multi_fitdist.R

```diff
@@ -1,133 +1,141 @@
-#' Maximum likelihood fitting of multiple distributions
+#' Helper function to call a fitting function across different models
 #'
-#' @param data A numeric vector of values
+#' @description This is a utility function which executes a function provided on
+#' the data provided and over all models specified. The function then organises
+#' the output and calculates the AIC and BIC and ranks the output by model fit
+#' which is determined by the `rank_by` argument.
+#'
+#' @details The vector of models given in the `models` argument needs to be
+#' named with the name of the model argument in the supplied function when it
+#' is not the second argument in that function. All elements of the vector need
+#' to be named, see example.
+#'
+#' The data is assumed to always be the first argument of the function supplied
+#' in `func`, `multi_fitdist()` will not work correct if this is not the case.
```

Comment:
> will not work correctly

We should explicitly state that it will throw an error.

Reply:
I am also thinking this is going to be common for most of the base R functions for model fitting like survival analysis `(?survival::survreg),` etc; the user will always have to wrap the function passed to `func`. We might want to consider a more flexible design. For example, if `func` takes in a list(func = some function, data = some data), we can manipulate `func` and assign the elements flexibly. Speaking of which, we will also have to accommodate formula-based models like `lm`, 'survival::survreg', etc.

Reply:
I agree it's a fine line between a trivial wrapper and providing some genuine utility. For now I think we should leave the structure of the function arguments as is, but I'm happy to revisit and we can make an issue to address this in the future. 

The same goes for the formula based models, I think this development should go in its own PR. 

Could you provide a reprex for `survival::survreg` using {quickfit}?

Reply:
> The same goes for the formula based models, I think this development should go in its own PR.

Agreed.

Reply:
> Could you provide a reprex for `survival::survreg` using {quickfit}?

Sure. `survival::survreg()` actually computes [two loglikelihoods](https://www.rdocumentation.org/packages/survival/versions/3.5-5/topics/survreg.object), one for the baseline and one for the full models. So, in this case, we get four loglikelihoods for two models. 

```
library(survival)

# Define fitting function

fit_survreg <- function(data, dist, ...) {
  survival::survreg(
    data = data,
    dist = dist,
    formula = Surv(futime, fustat) ~ ecog.ps + rx, # taken from ?survival::survreg
    scale = 1
  )
}

# Apply multi_fitdist

multi_fitdist(
  data = survival::ovarian,
  models = c(dist = "lognormal", dist = "weibull"),
  func = fit_survreg
)

# Results

     models  loglik.1  loglik.2      aic      bic
1   weibull -96.24333 -97.19804 196.4867 199.0029
2   weibull -96.24333 -97.19804 198.3961 200.9123
3 lognormal -97.74229 -98.03220 199.4846 202.0008
4 lognormal -97.74229 -98.03220 200.0644 202.5806

Created on 2023-05-19 with [reprex v2.0.2](https://reprex.tidyverse.org/)
```

Reply:
Thanks for the reprex. I will make an issue including the code chunk. We can discuss and make the development in a new PR.